README.md

```diff
@@ -155,6 +155,14 @@ az containerapp update \
   --image $(ko build ./cmd/app)
 ```
 
+Or [Oracle Cloud Functions](https://www.oracle.com/cloud-native/functions/)
+
+```
+fn update function \
+  my-container-app my-function-name \
+  --image $(ko build -P . | sed 's/@sha256:/:sha256-/')
```

Comment:
I don't believe this will work, since `ko` won't push to that tag. Have you confirmed this works in your environment?

Reply:
Yes, Oracle Cloud Functions expect following form for container image.

```
<region-key>.ocir.io/<tenancy-namespace>/<repo-name>/<func-name>:<tag>
```

https://docs.oracle.com/en-us/iaas/Content/Functions/Tasks/functionsupdatingfunctions.htm#notesdigest

Reply:
It sounds from reading this like OCF will accept image references by digest just fine, but that it won't automatically pull any new image with that tag until you update the function to specify the tag. Based on my reading, this is consistent with Cloud Run's behavior, and likely Lambda's, Fly.io's, etc.

Specifying images by digest is usually considered a best practice, so I'd like to recommend it here. This would also make our OCF recommendation consistent with the other deployment targets.

Can you test that this works when deploying by digest? And that the function is updated correctly when a new image by digest is specified in a subsequent call to `fn update function`? If that works, we should recommend that.

Reply:
```
fn update function my-app my-func --image $(ko bulid -P .)
```
This return error.
> Fn: Service error:InvalidParameter. Invalid Invalid Image - Image is not a valid docker image name. http status code: 400. Opc

And this too.
```
fn update function my-app my-func --image $(ko bulid .)
```

Oracle Cloud Function does not handle `registory-url/repo-name/func-name:tag-name@sha256:xxxxx` for `--image`.

Reply:
Okay, that's unfortunate. In that case you might need to specify `--bare` so the image gets pushed to exactly `KO_DOCKER_REPO`, without any additional path components.

Reply:
FWIW, it's also unclear from the docs whether the registry needs to be OCIR -- the docs seem to indicate that any registry can be used:

> (this documentation assumes you will be using [Oracle Cloud Infrastructure Registry](https://docs.oracle.com/en-us/iaas/Content/Registry/Concepts/registryoverview.htm#Overview_of_Registry) as your Docker registry and provides instructions accordingly).

(from https://docs.oracle.com/en-us/iaas/Content/Functions/Concepts/functionsprerequisites.htm)

Reply:
Okay, updated.

Reply:
If you want short history of commits, Please let me known. I will squash them.